antarest/study/storage/utils.py

```diff
@@ -390,3 +401,79 @@ def ignore_outputs(directory: str, _: t.Sequence[str]) -> t.Sequence[str]:
         study.tree.denormalize()
         duration = "{:.3f}".format(time.time() - stop_time)
         logger.info(f"Study '{study_dir}' denormalized in {duration}s")
+
+
+def is_folder_safe(workspace: WorkspaceConfig, folder: str) -> bool:
+    """
+    Check if the provided folder path is safe to prevent path traversal attack.
+
+    Args:
+        workspace: The workspace name.
+        folder: The folder path.
+
+    Returns:
+        `True` if the folder path is safe, `False` otherwise.
+    """
+    requested_path = workspace.path / folder
+    safe_dir = os.path.realpath(workspace.path)
+    real_requested_path = os.path.realpath(requested_path)  # resolve symbolic links like ~ , .. and .
+    # we check if real_requested_path starts with safe_dir,
+    # if not, it means that the requested path is outside the workspace
+    # TODO: replace by is_relative_to when we switch to python 3.11
+    common_prefix = os.path.commonprefix((real_requested_path, safe_dir))  # get the common prefix of the two paths
+    return common_prefix == safe_dir  # if requested path isn't in workspace, then common prefix is different
+
+
+def is_study_folder(path: Path) -> bool:
+    return path.is_dir() and (path / "study.antares").exists()
+
+
+def is_aw_no_scan(path: Path) -> bool:
+    return (path / "AW_NO_SCAN").exists()
+
+
+def get_workspace_from_config(config: Config, workspace_name: str, default_allowed: bool = False) -> WorkspaceConfig:
+    if not default_allowed and workspace_name == DEFAULT_WORKSPACE_NAME:
+        raise CannotAccessInternalWorkspace()
+    try:
+        return config.storage.workspaces[workspace_name]
+    except KeyError:
+        logger.error(f"Workspace {workspace_name} not found")
+        raise WorkspaceNotFound(f"Workspace {workspace_name} not found")
+
+
+def get_folder_from_workspace(workspace: WorkspaceConfig, folder: str) -> Path:
+    if not is_folder_safe(workspace, folder):
+        raise ValueError(f"Invalid path for folder: {folder}")
```

Comment:
Either raise a 422 error or catch the ValueError inside the explorer_service to raise a 422 cause for now it raises a 500 within the API and it's not great.